browsermob-core/src/main/java/net/lightbody/bmp/proxy/selenium/SeleniumProxyHandler.java

```diff
@@ -637,8 +634,10 @@ protected void wireUpSslWithCyberVilliansCA(String host, SslRelay listener) {
       }
 
       public void cleanSslWithCyberVilliansCA(){
-          if(deleteDirectoryTask != null) {
-              deleteDirectoryTask.run();
+          if(!deleteDirectoryTasks.isEmpty()) {
```

Comment:
Put this entire block into synchronized (deleteDirectoryTasks) to avoid multi-threading issues.